src/main/kotlin/build/buf/gradle/ProtobufGradlePluginSupport.kt

```diff
@@ -102,39 +102,63 @@ private fun Task.workspaceCommonConfig() {
+        .joinToString("\n") { "|  - $it" }
+
+// Returns all directories that have may have proto files relevant to processing the project's proto files. This
+// includes any proto files that are simply references (includes) as well as those that will be processed (code
+// generation or validation).
+private fun Task.allProtoDirs(): List<File> =
+    (project.allProtoSourceSetDirs() + project.file(Paths.get(BUILD_EXTRACTED_INCLUDE_PROTOS_MAIN))).filter { anyProtos(it) }
+
+// Returns the list of directories containing proto files defined in *this* project. The returned directories do *not*
+// include those that contain protos defined in dependencies and placed in the extracted-protos directory for codegen.
+// The returned directories *only* contain the proto files that should be processed by buf for operations like linting,
+// format checking, and breaking change detection.
+//
+// NOTE: We explicitly remove BUILD_EXTRACTED_PROTOS_MAIN from the list of source set directories in order to fix issue
+// https://github.com/bufbuild/buf-gradle-plugin/issues/132. Starting in version 0.9.2 of
+// protobuf-gradle-plugin, the "proto" source set srcDirs includes the outputs of the "extractedProtos" task, which
+// breaks this plugin when used with the protobuf-gradle-plugin and "protobuf" dependencies (which generates code for
+// any proto files within those dependencies).
+//
+// Protobuf-gradle-plugin change that introduced this behavior: https://github.com/google/protobuf-gradle-plugin/pull/637/
+// Line: https://github.com/google/protobuf-gradle-plugin/blob/9d2a328a0d577bf4439d3b482a953715b3a03027/src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy#L425
+internal fun Project.projectDefinedProtoDirs() =
+    allProtoSourceSetDirs() - file(Paths.get(BUILD_EXTRACTED_PROTOS_MAIN))
 
-internal fun Project.srcProtoDirs() =
-    the<SourceSetContainer>().flatMap { it.protoDirs(this) } + androidSrcProtoDirs()
+// Returns deduplicated list of all proto source set directories.
+private fun Project.allProtoSourceSetDirs() = (projectProtoSourceSetDirs() + androidProtoSourceSetDirs()).toSet().toList()
 
-private fun Project.androidSrcProtoDirs() =
+// Returns android proto source set directories that protobuf-gradle-plugin will codegen.
+private fun Project.androidProtoSourceSetDirs() =
     extensions.findByName("android")
         ?.let { baseExtension ->
             val prop = baseExtension::class.declaredMemberProperties.single { it.name == "sourceSets" }
             @Suppress("UNCHECKED_CAST")
             (prop as KProperty1<Any, Set<ExtensionAware>>).get(baseExtension)
         }
         .orEmpty()
-        .flatMap { it.protoDirs(this) }
+        .flatMap { it.projectProtoSourceSetDirs() }
+
+// Returns all proto source set directories that the protobuf-gradle-plugin will codegen.
+private fun Project.projectProtoSourceSetDirs() = the<SourceSetContainer>().flatMap { it.projectProtoSourceSetDirs() }
```

Comment:
Tiny nit: for consistency with the rest of the project and the [rectangle rule](https://github.com/google/google-java-format/wiki/The-Rectangle-Rule) for when expression functions expand to be multi-line, a line break for expression functions:
```suggestion
private fun Project.projectProtoSourceSetDirs() =
    the<SourceSetContainer>().flatMap { it.projectProtoSourceSetDirs() }
```